webpki-bin/src/bin/webpki.rs

```diff
@@ -0,0 +1,138 @@
+use clap::Clap;
+use std::convert::TryFrom;
+use std::env;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process;
+use std::time::SystemTime;
+
+#[derive(Clap)]
+#[clap(about = "Parse and print a certificate")]
+struct PrintCert {
+    filename: PathBuf,
+}
+
+#[derive(Clap)]
+#[clap(about = "Verify server certificate")]
+struct VerifyServerCert {
+    #[clap(long)]
+    trusted_root: PathBuf,
+    #[clap(long)]
+    intermediates: Vec<PathBuf>,
+    #[clap(long)]
+    server_cert: PathBuf,
+    #[clap(long, about = "Seconds since epoch")]
+    time: Option<u64>,
+}
+
+#[derive(Clap)]
+enum Command {
+    PrintCert(PrintCert),
+    VerifyServerCert(VerifyServerCert),
+}
+
+#[derive(Clap)]
+#[clap(about = "Utility to debug webpki. Does not have a stable API.")]
+struct Opts {
+    #[clap(subcommand)]
+    command: Command,
+}
+
+fn print_error_and_exit(s: String) -> ! {
+    let arg0 = env::args().next();
+    let arg0 = arg0.as_ref().map(|s| s.as_str()).unwrap_or("webpki");
+    eprintln!("{}: {}", arg0, s);
+    process::exit(1)
+}
+
+fn read_file(path: &Path) -> Vec<u8> {
+    match fs::read(path) {
+        Ok(bytes) => bytes,
+        Err(e) => print_error_and_exit(format!("could not read file {}: {}", path.display(), e)),
+    }
+}
+
+fn parse_end_entity_cert<'a>(der: &'a [u8], path: &Path) -> webpki::EndEntityCert<'a> {
+    match webpki::EndEntityCert::try_from(der) {
+        Ok(cert) => cert,
+        Err(e) => {
+            print_error_and_exit(format!("failed to parse a cert {}: {}", &path.display(), e))
+        }
+    }
+}
+
+fn print_cert(args: PrintCert) {
+    let der = read_file(&args.filename);
+
+    let _cert = parse_end_entity_cert(&der, &args.filename);
+
+    eprintln!("cert {} parsed successfully", &args.filename.display());
+    // TODO: actually print something about the certificate
+}
+
+static ALL_SIGALGS: &[&webpki::SignatureAlgorithm] = &[
```

Comment:
I think this should be something the webpki library exports.

Reply:
#192